src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+      .Set(blanket_incommod)
+      .Start();
+
+  // Tritium Buy Policy Selection:
+  if (refuel_mode == "schedule") {
+    cyclus::IntDistribution::Ptr active_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(1));
+    cyclus::IntDistribution::Ptr dormant_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(buy_frequency - 1));
+    cyclus::DoubleDistribution::Ptr size_dist =
+        cyclus::FixedDoubleDist::Ptr(new cyclus::FixedDoubleDist(1));
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dist)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              std::string("ss"), reserve_inventory, reserve_inventory)
+        .Set(fuel_incommod);
+  } else {
+    throw cyclus::KeyError("Refill mode " + refuel_mode +
+                           " not recognized! Try 'schedule' or 'fill'.");
+    RecordOperationalInfo("Transaction Error",
+                          "Refill mode " + refuel_mode +
+                              " not recognized! Try 'schedule' or 'fill'.");
+  }
+
+  tritium_sell_policy
+      .Init(this, &tritium_excess, std::string("Excess Tritium"))
+      .Set(fuel_incommod)
+      .Start();
+  helium_sell_policy.Init(this, &helium_storage, std::string("Helium-3"))
+      .Set(he3_outcommod)
+      .Start();
+}
+
+std::string Reactor::GetComp(cyclus::Material::Ptr mat) {
+  std::string comp = "{";
+  cyclus::CompMap c = mat->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+  for (std::map<const int, double>::const_iterator it = c.begin();
+       it != c.end(); ++it) {
+    comp = comp + std::string("{") + std::to_string(it->first) +
+           std::string(",") + std::to_string(it->second) + std::string("},");
+  }
+  comp.pop_back();
+  comp = comp + std::string("}");
+  return comp;
+}
+
+void Reactor::SequesterTritium(){
+  if (!tritium_sequestered.empty()){
+    cyclus::Material::Ptr sequestered_mat = tritium_sequestered.Pop();
+    cyclus::CompMap c = sequestered_mat->comp()->atom();
+    cyclus::compmath::Normalize(&c, sequestered_mat->quantity());
+
+    double equilibrium_deficit = sequestered_equilibrium - c[tritium_id];
+
+    // Another catch might be good here...
+    sequestered_mat->Absorb(tritium_storage.Pop(equilibrium_deficit));
+
+    tritium_sequestered.Push(sequestered_mat);
+
+    
+  } else {
+    // Should I do a try/catch here, or is "user error" fine?
+    tritium_sequestered.Push(tritium_storage.Pop(sequestered_equilibrium));
+  }
+}
+
+void Reactor::Startup() {
+  cyclus::Material::Ptr initial_storage = tritium_storage.Peek();
+  cyclus::CompMap c = initial_storage->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+
+  if ((tritium_storage.quantity() >= (startup_inventory)) &&
```

Comment:
Since all other branches result in throwing exceptions, do those first one at a time:
* not enough fuel
* less than usage
* wrong comp

then if you get through those, just do startup stuff

Reply:
Yep, that's absolutely the right way to do that. Good point. Fixed